server/lib/routes/get-index.js

```diff
@@ -9,9 +25,44 @@ module.exports = function () {
   route.path = '/';
 
   route.process = function (req, res) {
-    res.render('index');
+    if (! shouldEmitActivityEvent()) {
+      return res.render('index');
+    }
+
+    var time = Date.now();
+    var flowId = crypto.randomBytes(16).toString('hex');
+    res.render('index', { flowBeginTime: time, flowId: flowId });
+
+    setImmediate(function () {
+      var userAgent = ua.parse(req.headers['user-agent']);
+      // This event data is structured to be consistent with
+      // the auth server activity events.
+      var activityEvent = {
+        /*eslint-disable camelcase*/
+        device_id: null,
+        event: 'flow.begin',
+        flow_id: flowId,
+        flow_time: 0,
+        new_user: false,
+        time: time,
+        uid: null,
+        user_agent_browser: userAgent.ua.family,
+        user_agent_os: userAgent.os.family + ' ' + userAgent.os.major,
+        user_agent_version: userAgent.ua.major
+        /*eslint-enable camelcase*/
+      };
+      activityEventQueryParams.forEach(function (key) {
+        activityEvent[key] = req.query[key] || null;
+      });
+      // The data pipeline only reads from stderr.
+      process.stderr.write('activityEvent ' + JSON.stringify(activityEvent) + '\n');
```

Comment:
The fields look good, thanks @philbooth.  To make it easy to process with in our metrics pipeline, let's make sure we're emitting it in the same format as events from other servers, i.e. some variant of the "mozlog" [1] JSON format, which I *think* is the one documented at [2].  It may even be worth trying to use mozlog directly since we're in the server here.

[1] https://github.com/mozilla/mozlog
[2] https://mana.mozilla.org/wiki/pages/viewpage.action?pageId=42895640